src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchTrackingService.java

```diff
@@ -167,20 +166,11 @@ public void setup(FloodlightModuleContext context) {
         }
     }
 
-    private void dumpAllSwitchesAction(String correlationId) {
+    private void dumpAllSwitchesAction() {
         Collection<IOFSwitch> iofSwitches = switchManager.getAllSwitchMap(true).values();
-        int switchCounter = 0;
         for (IOFSwitch sw : iofSwitches) {
-            try {
-                NetworkDumpSwitchData swData = new NetworkDumpSwitchData(buildSwitch(sw));
-                producerService.sendMessageAndTrack(discoveryTopic,
-                                                correlationId,
-                                                new ChunkedInfoMessage(swData, System.currentTimeMillis(),
-                                                        correlationId, switchCounter, iofSwitches.size(), region));
-            } catch (Exception e) {
-                logger.error("Failed to send network dump for {}", sw.getId());
-            }
-            switchCounter++;
+            NetworkDumpSwitchData payload = new NetworkDumpSwitchData(buildSwitch(sw));
+            emitDiscoveryEvent(sw.getId(), payload);
```

Comment:
nit: Why is "emit discovery event" used? This is not related to it.

Reply:
I can't produce a better name at this moment. Can you propose some? `emitDiscoveryEvent` is used to send any event into network-topology. Most of them are "discovery" events, and only this one "dump" is not exactly discovery event. But from the consumer perspective the switch-dump is a discovery message too. So I have named it as `emitDiscoveryEvent`.

PS I open to propositions.